### PR TITLE
GRW-1716 - refact: extract 'insurances by market' logic into a service

### PR DESCRIPTION
**Pull request by guilhermespopolin**
📓 This is part of a series of PRs; I believe it's easier to review then Top to Bottom;

## Describe your changes

* Refactor `embark` service so its API function in terms of `Market` instead of `MarketLabel`; In order to this to work, I had to include `Market` information in the `LocaleData` record (`src/lib/l10n/locales.ts`).
* Creates an `insurances` service responsible to handle what it was being handled by `LandingPage.constants`/`LandingPage.helpers` before: provides an API (`getInsurancesByLocaleLabel`) to get insurances by locale (in other terms, per market).
* Add a _graphql_ query for fetching coverage info of an insurance. That info is included in the insurance data provided by the `insurances` service (previous item).

## Justify why they are needed

It simplifies things and provides a data requirement for the [`CoverageDialog`](https://github.com/HedvigInsurance/racoon/blob/GRW-1716/feat/insurance-card/apps/onboarding/src/components/LandingPage/CoverageDialog.tsx)

## Preview (End result)

[onboarding DK](https://onboarding-git-grw-1716-featinsurance-card-hedvig.vercel.app/dk-en/new-member)
[onboarding NO](https://onboarding-git-grw-1716-featinsurance-card-hedvig.vercel.app/no-en/new-member)

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #834** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/834" target="_blank">Graphite</a> 
    * **PR #835** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/835" target="_blank">Graphite</a> 
      * **PR #836** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/836" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/836?utm_source=stack-comment).

File: apps/onboarding/src/services/insurances/index.ts [hunk 1]
```diff
@@ -1,8 +1,28 @@
+import { getLocale } from '@/lib/l10n'
+import { LocaleLabel } from '@/lib/l10n/locales'
 import { Market, MarketLabel } from '@/lib/types'
 import { Features, Feature } from '@/services/features'
-import { Insurances } from './LandingPage.types'
+import { TypeOfContract } from '@/services/graphql/generated'
+import { graphqlSdk } from '@/services/graphql/sdk'
 
-export const INSURANCES_BY_MARKET: Record<Market, Insurances> = {
+export type Insurance = {
```
Comment (robinandeer) [inline apps/onboarding/src/services/insurances/index.ts:8]: I would try to come up with a less generic name if possible -- but no need to spend too much time on it for this part of the code base.

File: apps/onboarding/src/services/insurances/index.ts [hunk 2]
```diff
@@ -128,9 +157,29 @@ export const INSURANCES_BY_MARKET: Record<Market, Insurances> = {
...
 }
+
+export const Insurances = {
+  async getInsurancesByLocaleLabel(localeLabel: LocaleLabel) {
+    const locale = getLocale(localeLabel)
+    const insurancesWithouthPerils = INSURANCES_BY_MARKET[locale.market]
+
+    const insurances: Array<Insurance> = []
+    for (const insurance of insurancesWithouthPerils) {
+      const perilsData = await graphqlSdk.ContractPerils({
```
Comment (robinandeer) [inline apps/onboarding/src/services/insurances/index.ts:175]: This is slower than it has to be -- we should consider running it in parallel using `Promise.all` instead.

File: apps/onboarding/src/pages/new-member/index.tsx [hunk 3]
```diff
@@ -55,7 +52,7 @@ export const getServerSideProps: GetServerSideProps<LandingPageProps> = async (c
   }
 
   const locale = context.locale as LocaleLabel
-  const insurances = getInsurancesByLocaleLabel(locale)
+  const insurances = await Insurances.getInsurancesByLocaleLabel(locale)
```
Comment (robinandeer) [inline apps/onboarding/src/pages/new-member/index.tsx:55]: Should we try catch this just to be safe?

**Review by robinandeer — APPROVED** [review]
Looks good, just posted some improvement comments

File: apps/onboarding/src/services/insurances/index.ts (hunk 1 above)
Comment (guilhermespopolin) [reply apps/onboarding/src/services/insurances/index.ts:8]: You mean something like product?

**Pull request merged**